### PR TITLE
Makes possibile to ignore tasks

### PR DESCRIPTION
**Pull request by mfrascati**
Implements #319

### CODE REVIEW

File: config/app_queue.php [hunk 1]
```diff
@@ -62,5 +62,8 @@
 
 		// Additional plugins to include tasks from (if they are not already loaded anyway)
 		'plugins' => [],
+
+		// ignores task classes
+		'ignoredtasks' => [],
```
Comment (dereuromark) [inline config/app_queue.php:67]:
```suggestion
		'ignoredTasks' => [],
```

File: src/Queue/Config.php [hunk 2]
```diff
@@ -64,6 +64,18 @@ public static function maxworkers() {
 		return Configure::read('Queue.maxworkers', 1);
 	}
 
+	/**
+	 * @return array
+	 */
+	public static function ignoredtasks() {
```
Comment (dereuromark) [inline src/Queue/Config.php:70]: same here
You can also add return type :array here.

File: src/Queue/TaskFinder.php [hunk 3]
```diff
@@ -86,7 +87,10 @@ protected function getTasks(string $path, ?string $plugin = null): array {
 			/** @phpstan-var class-string<\Queue\Queue\Task> $className */
 			$className = $namespace . '\Queue\Task\\' . $name . 'Task';
 			$key = $plugin ? $plugin . '.' . $name : $name;
-			$tasks[$key] = $className;
+
+			if (!in_array($className, $ignoredTasks)) {
```
Comment (dereuromark) [inline src/Queue/TaskFinder.php:91]:
```suggestion
			if (!in_array($className, $ignoredTasks, true)) {
```

File: config/app_queue.php (hunk 1 above)
Comment (mfrascati) [reply config/app_queue.php:67]: yeah I wasn't sure about which style to adopt since other options are a bit mixed :)

File: src/Queue/Config.php [hunk 4]
```diff
@@ -64,6 +64,18 @@ public static function maxworkers() {
 		return Configure::read('Queue.maxworkers', 1);
 	}
 
+	/**
+	 * @return array
+	 */
+	public static function ignoredtasks(): array {
```
Comment (dereuromark) [inline src/Queue/Config.php:70]: This is still not camelCase

File: src/Queue/Config.php [hunk 5]
```diff
@@ -64,6 +64,18 @@ public static function maxworkers() {
...
+		$a = Configure::read('Queue.ignoredtasks', []);
```
Comment (dereuromark) [inline src/Queue/Config.php:71]: Same here

File: src/Queue/Config.php [hunk 6]
```diff
@@ -64,6 +64,18 @@ public static function maxworkers() {
...
+		if (!is_array($a)) {
+			throw new InvalidArgumentException('Queue.ignoredtasks is not an array');
```
Comment (dereuromark) [inline src/Queue/Config.php:73]: Or here

File: src/Queue/Config.php (hunk 4 above)
Comment (mfrascati) [reply src/Queue/Config.php:70]: Missed the file, sorry

**Pull request merged**